R/init.R

```diff
@@ -4,44 +4,43 @@
 # error messages.
 
 
-#' Create the Server and UI Function For the Shiny App
+#' Create the server and UI function for the `shiny` app
 #'
 #' @description `r lifecycle::badge("stable")`
+#'
 #' End-users: This is the most important function for you to start a
-#' teal app that is composed out of teal modules.
+#' `teal` app that is composed out of `teal` modules.
 #'
 #' @details
 #' When initializing the `teal` app, if `datanames` are not set for the `teal_data` object,
 #' defaults from the `teal_data` environment will be used.
 #'
-#' @param data (`teal_data`, `teal_data_module`, `named list`)\cr
-#' `teal_data` object as returned by [teal.data::teal_data()] or
-#' `teal_data_module` or simply a list of a named list of objects
-#' (`data.frame` or `MultiAssayExperiment`).
-#' @param modules (`list`, `teal_modules` or `teal_module`)\cr
+#' @param data (`teal_data` or `teal_data_module`)
+#' `teal_data` object as returned by [`teal.data::teal_data()`] or `teal_data_module`.
```

Comment:
```suggestion
#' `teal_data` object as returned by [`teal.data::teal_data()`] or [`teal_data_module()`].
```

Reply:
`teal_data_module` does not return a `teal_data` object. Either leave this as is or change to "or a `teal_data_module`'s server function".

Reply:
Actually, line 19 may be considered redundant.

Reply:
Hmmm I think it would be good from our side to add references to class constructors / wrappers if custom-built class is required. I believe this would be very helpful to the end-users. Especially for `init()` which oftentimes is the very first package function end-users are facing.
ATM this applies for both `data` and `filters`. `module` argument has references to `module()` and `modules()` and that's good.

Reply:
I agree, we had references earlier and I have added them back again. I have slightly changed the param description. Let me know your thoughts. Otherwise, I will add ref as just `see [<topic>].`
[d432ec2](https://github.com/insightsengineering/teal/pull/1021/commits/d432ec27e80061e6583c8f5f06b9c626d0bb8150)

Reply:
It's good now! Thanks 👍